### PR TITLE
Unsubscribe on partition reassignment

### LINKED ISSUE #882: Running zilla with the `kafka-grpc` binding in a cluster with multiple instances results in each instance delivering a message to the configured `remote_server`
**Describe the bug**
Running 3 zilla instances with the exact same config delivering messages to a gRPC service using the `kafka-grpc` binding results in each zilla instance delivering the same message.

**To Reproduce**
1. run the [grpc.kafka.proxy](https://github.com/aklivity/zilla-examples/tree/main/grpc.kafka.proxy) example with replica count of 3
2. 3 messages are delivered to the gRPC service

**Expected behavior**
Zilla should respect the consumer group rules and distribute messages across zilla instances. (n) number of zilla instances should work with (n) number of partitions for the same topic.

### PR DESCRIPTION
**Pull request by akrambek**
## Description

Unsubscribe on partition reassignment

Fixes #882

### CODE REVIEW

File: runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/KafkaCacheServerConsumerFactory.java [hunk 1]
```diff
@@ -533,7 +536,8 @@ final class KafkaCacheServerConsumerFanout
         private final long routedId;
         private final long authorization;
         private final ArrayList<KafkaCacheServerConsumerStream> streams;
-        private final Object2ObjectHashMap<String, String> members;
+        private final Map<String, String> members;
+        private final ObjectHashSet<String> tempMembers;
```
Comment (jfallows) [inline runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/KafkaCacheServerConsumerFactory.java:540]:
```suggestion
        private final ObjectHashSet<String> memberIds;
```

File: runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/KafkaCacheServerConsumerFactory.java [hunk 2]
```diff
@@ -578,7 +582,8 @@ private KafkaCacheServerConsumerFanout(
             this.groupId = groupId;
             this.timeout = timeout;
             this.streams = new ArrayList<>();
-            this.members = new Object2ObjectHashMap<>();
+            this.tempMembers = new ObjectHashSet<>();
+            this.members = new LinkedHashMap<>();
```
Comment (jfallows) [inline runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/KafkaCacheServerConsumerFactory.java:586]: Are we switching to `LinkedHashMap` to maintain insertion order, or for `putIfAbsent`?
Note: putIfAbsent is also available on `Object2ObjectHashMap`, inherited from `Map`.

File: runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/KafkaClientOffsetFetchFactory.java [hunk 3]
```diff
@@ -940,7 +943,7 @@ private void doApplicationBegin(
         private void doApplicationData(
             long traceId,
             long authorization,
-            KafkaDataExFW extension)
+            Flyweight extension)
```
Comment (jfallows) [inline runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/KafkaClientOffsetFetchFactory.java:946]: Is this change necessary?

**Review by jfallows — CHANGES_REQUESTED** [review]

File: runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/KafkaCacheServerConsumerFactory.java (hunk 2 above)
Comment (akrambek) [reply runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/KafkaCacheServerConsumerFactory.java:586]: to maintain insertion order

**Review by jfallows — APPROVED** [review]

**Pull request merged**